t/io_uring.c

```diff
@@ -841,18 +842,22 @@ static struct submitter *get_submitter(int offset)
 	return ret;
 }
 
-static void sig_int(int sig)
+static void do_finish()
```

Comment:
Maybe take a const char * argument to do_finish() and have it print

Exiting on signal

for signals, and

Exiting on timeout

for non-signal exits. Don't need to print the signal, just get rid of that. And with that change, my other comment would no longer apply - you're missing a void here.